src/main/java/Du/Parser.java

```diff
@@ -0,0 +1,117 @@
+package Du;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Parser {
+    private TaskList tasks;
+
+    /**
+     * public constructor for Parser
+     * @param tasks
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * takes in commands from user and parse them
+     */
+    public void parse() {
```

Comment:
I think the responsibility of Parser should only be parsing 1 command at a time, and the loop should be in Duke class. What do you think?